scribejava-apis/src/test/java/com/github/scribejava/apis/examples/Twitter20WithPKCEExample.java

```diff
@@ -0,0 +1,133 @@
+package com.github.scribejava.apis.examples;
+
+import java.util.Random;
+import java.util.Scanner;
+
+import com.github.scribejava.apis.TwitterApi20;
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.oauth.AuthorizationUrlBuilder;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.model.OAuthRequest;
+import com.github.scribejava.core.model.Response;
+import com.github.scribejava.core.model.Verb;
+import com.github.scribejava.core.oauth.AccessTokenRequestParams;
+import com.github.scribejava.core.oauth.OAuth20Service;
+import com.github.scribejava.core.revoke.TokenTypeHint;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+public class Twitter20WithPKCEExample {
+
+    private static final String NETWORK_NAME = "Twitter";
+    private static final String PROTECTED_RESOURCE_URL = "https://api.twitter.com/2/tweets?ids=1261326399320715264,1278347468690915330";
+
+    private Twitter20WithPKCEExample() {
+    }
+
+    @SuppressWarnings("PMD.SystemPrintln")
+    public static void main(String... args) throws IOException, InterruptedException, ExecutionException {
+        final String clientId = "CLIENT_ID"; // replace these with your client id
+        final String state = "secret" + new Random().nextInt(999_999);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [DMI: Random object created and used only once (DMI_RANDOM_USED_ONLY_ONCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dmi-random-object-created-and-used-only-once-dmi-random-used-only-once).
This code creates a java.util.Random object, uses it to generate one random number, and then discards the Random object. This produces mediocre quality random numbers and is inefficient. If possible, rewrite the code so that the Random object is created once and saved, and each time a new random number is required invoke a method on the existing Random object to obtain it.
If it is important that the generated Random numbers not be guessable, you *must* not create a new Random for each random number; the values are too easily guessable. You should strongly consider using a java.security.SecureRandom instead (and avoid allocating a new SecureRandom for each random number needed).